### DISCUSSION TITLE
Changing context style in setup is not reflected in update

### DISCUSSION THREAD

**Discussion by miam-miam** [emilk/egui#630] [Q&A] [answered]
Hi I am trying to remove the shadow on a window and am unable to do it from setup, the style is only changed if I put the code in update. Is this normal or am I doing something wrong?

```rust
impl epi::App for CalcApp {
    fn update(&mut self, ctx: &egui::CtxRef, frame: &mut epi::Frame<'_>) {
        Window::new("Calculator")
            .frame(Frame::window(&ctx.style())) // Is still using default style
            .show(ctx, |ui| {
                ui.label("This is a window!");
            });
    }

    fn setup(
        &mut self,
        ctx: &egui::CtxRef,
        _frame: &mut epi::Frame<'_>,
        _storage: Option<&dyn epi::Storage>,
    ) {
        let mut style: egui::Style = (*ctx.style()).clone();
        style.visuals.window_shadow = Shadow {
            extrusion: 0.0,
            color: color::Color32::TRANSPARENT,
        };
        ctx.set_style(style);
    }
}
```

**Comment by emilk** [comment, marked as answer]
I fail to reproduce the problem on latest `master`.

**Reply by miam-miam** [reply]
Thanks for checking it out was using eframe on Windows, will try to reproduce it using egui_glium and report back!

**Reply by emilk** [reply]
`egui_glium` is the native backend of `eframe`. If you wanna try on latest bleeding edge, add this to `Cargo.toml`:

``` toml
[patch.crates-io]
eframe = { git = "https://github.com/emilk/egui", branch = "master" }
```

**Reply by miam-miam** [reply]
Fixed thanks!

**Reply by miam-miam** [reply]
Turns out as I was re-adding features that persistance is actually the root cause of the problem if you add `#[cfg(feature = "persistence")]` to the setup function (with `persistence = ["eframe/persistence", "serde"]` in .toml). The shadows reappear, is this normal?
